src/nvim/os/env.c

```diff
@@ -76,6 +76,31 @@ char *os_getenvname_at_index(size_t index)
   return name;
 }
 
+/// Check if the environment variable `name` exists and has
+/// a value of either 1, "on", "yes", or "true".
+bool os_env_istruthy(const char *name) FUNC_ATTR_NONNULL_ALL
+{
+  const char *val = os_getenv(name);
+  if (val == NULL || *val == NUL) {  // TODO(Pyrohh): Remove the check for
+    return false;                    // NUL after #2574 is merged.
+  }
+
+  errno = 0;
+  long n = strtol(val, NULL, 10);
```

Comment:
`val` points to the start of the string, so checking `*val == NUL` only checks the zeroth char. I assume it is not possible for `os_getenv` to return a non-NUL terminated string, but it hasn't been updated to support MSVC yet...